assets/src/block-editor/components/amp-preview.js

```diff
@@ -0,0 +1,318 @@
+		// navigation occurs or a new preview window is opened, to avoid
+		// unintentional forceful redirects.
+		if ( previewLink && ! prevProps.previewLink ) {
+			this.setPreviewWindowLink( previewLink );
+		}
+
+		this.moveButton();
+	}
+
+	/**
+	 * Moves the (non-AMP) 'Preview' button to before this 'Preview AMP' button, if it's not there already.
+	 */
+	moveButton() {
+		const buttonWrapper = get( this, [ 'buttonRef', 'current', 'parentNode' ], false );
+		if ( ! buttonWrapper ) {
+			return;
+		}
+
+		if ( ! buttonWrapper.previousSibling || ! buttonWrapper.previousSibling.classList.contains( POST_PREVIEW_CLASS ) ) {
+			const postPreviewButton = document.querySelector( `.${ POST_PREVIEW_CLASS }` );
+			if ( get( postPreviewButton, 'nextSibling' ) ) {
+				buttonWrapper.parentNode.insertBefore( buttonWrapper, postPreviewButton.nextSibling );
+			}
+		}
+	}
+
+	/**
+	 * Sets the preview window's location to the given URL, if a preview window
+	 * exists and is not closed.
+	 *
+	 * @param {string} url URL to assign as preview window location.
+	 */
+	setPreviewWindowLink( url ) {
+		const { previewWindow } = this;
+
+		if ( previewWindow && ! previewWindow.closed ) {
+			previewWindow.location = url;
+		}
+	}
+
+	/**
+	 * Gets the window target.
+	 */
+	getWindowTarget() {
+		const { postId } = this.props;
+		return `amp-preview-${ postId }`;
+	}
+
+	/**
+	 * Opens the preview window.
+	 *
+	 * @param {Object} event The DOM event.
+	 */
+	openPreviewWindow( event ) {
+		// Our Preview button has its 'href' and 'target' set correctly for a11y
+		// purposes. Unfortunately, though, we can't rely on the default 'click'
+		// handler since sometimes it incorrectly opens a new tab instead of reusing
+		// the existing one.
+		// https://github.com/WordPress/gutenberg/pull/8330
+		event.preventDefault();
+
+		// Open up a Preview tab if needed. This is where we'll show the preview.
+		if ( ! this.previewWindow || this.previewWindow.closed ) {
+			this.previewWindow = window.open( '', this.getWindowTarget() );
+		}
+
+		// Focus the Preview tab. This might not do anything, depending on the browser's
+		// and user's preferences.
+		// https://html.spec.whatwg.org/multipage/interaction.html#dom-window-focus
+		this.previewWindow.focus();
+
+		// If we don't need to autosave the post before previewing, then we simply
+		// load the Preview URL in the Preview tab.
+		if ( ! this.props.isAutosaveable ) {
+			this.setPreviewWindowLink( event.target.href );
+			return;
+		}
+
+		// Request an autosave. This happens asynchronously and causes the component
+		// to update when finished.
+		if ( this.props.isDraft ) {
+			this.props.savePost( { isPreview: true } );
+		} else {
+			this.props.autosave( { isPreview: true } );
+		}
+
+		// Display a 'Generating preview' message in the Preview tab while we wait for the
+		// autosave to finish.
+		writeInterstitialMessage( this.previewWindow.document );
+	}
```

Comment:
Since this and `setPreviewWindowLink` are unchanged from core, could the core function not be called directly here?

Reply:
For example, this function could be:

```js
import { PostPreviewButton } from '@wordpress/editor';

// ...

openPreviewWindow( event ) {
    PostPreviewButton.prototype.openPreviewWindow.call( this, event )
}
```

Reply:
Alas, this doesn't seem to work. It seems `compose` prevents accessing the underlying functions on the prototype.